email: allow sending emails from EU Mailgun domains

freesteph opened the pull request:
Closes https://github.com/writefreely/writefreely/issues/944 

From the commit:

```
The Mailgun API requires being told when a EU email domain is used[1] so introduce a new EU_MAILGUN environement variable to do just that.

[1]: https://github.com/mailgun/mailgun-go?tab=readme-ov-file#usage
```

I've never written Go so do provide guidance and let me know what else needs to be done. I'm going to write a bit of documentation shortly (in `writefreely/documentation`). Thanks!

---

- [x] I have signed the [CLA](https://todo.musing.studio/L1)

thebaer commented:
Hey, thanks for bringing this up! Definitely want to support it.

For this, instead of grabbing straight from `os. GetEnv`, you'll want to add this value in `config/config.go`, and use that struct value. Would also recommend building off the changes in #1131 (base off the `smtp` branch), as we're moving things around with email functionality right now.

Let me know if you have any other questions!

freesteph committed:
email: allow sending emails from EU Mailgun domains

The Mailgun API requires being told when a EU email domain is used[1]
so introduce a new `mailgun_europe` config parameter to indicate that.

[1]: https://github.com/mailgun/mailgun-go?tab=readme-ov-file#usage

freesteph commented:
Hey, thanks for the feedback! I've rebased against the `smtp` branch and went through the config instead. Note [there's a bit in `email.go` that still uses the Mailgun constructor](https://github.com/writefreely/writefreely/blob/smtp/email.go#L311) directly rather than the new mail client, I don't know if that's intended or not.

freesteph commented:
gentle tis-the-season ping @thebaer 🎄

freesteph commented:
bump

thebaer commented:
Yes, I believe the remaining Mailgun stuff is addressed in #1153. But this looks great! Thanks again for tackling this. Merging now.
